File: app/src/main/java/yc/com/soundmark/index/contract/IndexContract.java
package yc.com.soundmark.index.contract;

import yc.com.base.IPresenter;
import yc.com.base.IView;
import yc.com.soundmark.index.model.domain.ContactInfo;

/**
 * Created by wanglin  on 2018/10/29 09:10.
 */
public interface IndexContract {

    interface View extends IView {
        void showIndexInfo(ContactInfo contactInfo);
    }

    interface Presenter extends IPresenter {
        void getIndexInfo();
    }
}

File: app/src/main/java/yc/com/homework/index/contract/IndexContract.kt
package yc.com.homework.index.contract

import yc.com.base.INoNet
import yc.com.base.IPresenter
import yc.com.base.IView
import yc.com.homework.index.domain.bean.IndexInfo
import yc.com.homework.index.domain.bean.NewsInfo

/**
 *
 * Created by wanglin  on 2019/2/15 14:34.
 */
interface IndexContract {
    interface View : IView, INoNet {
        fun showIndexInfo(t: IndexInfo?)
        fun showIndexRecommendInfos(t: ArrayList<NewsInfo>, fresh: Boolean)
    }

    interface Presenter : IPresenter {
        fun getIndexInfo()
    }
}